Selenium.Axe/Selenium.Axe/AxeBuilder.cs

```diff
@@ -11,18 +14,81 @@ namespace Selenium.Axe
     public class AxeBuilder
     {
         private readonly IWebDriver _webDriver;
-        private readonly IncludeExcludeManager _includeExcludeManager = new IncludeExcludeManager();
 
-        private static readonly AxeBuilderOptions DefaultOptions = new AxeBuilderOptions {ScriptProvider = new EmbeddedResourceAxeProvider()};
+        private readonly AxeRunContext runContext = new AxeRunContext();
+        private readonly AxeRunOptions runOptions = new AxeRunOptions();
+        private static readonly AxeBuilderOptions DefaultOptions = new AxeBuilderOptions { ScriptProvider = new EmbeddedResourceAxeProvider() };
+
+        private static readonly JsonSerializerSettings JsonSerializerSettings = new JsonSerializerSettings
+        {
+            Formatting = Formatting.None,
+            NullValueHandling = NullValueHandling.Include
+        };
 
         public string Options { get; set; } = "{}";
 
         /// <summary>
         /// Initialize an instance of <see cref="AxeBuilder"/>
         /// </summary>
         /// <param name="webDriver">Selenium driver to use</param>
-        public AxeBuilder(IWebDriver webDriver): this(webDriver, DefaultOptions)
+        public AxeBuilder(IWebDriver webDriver) : this(webDriver, DefaultOptions)
+        {
+        }
+
+        /// <summary>
+        /// Limit analysis to only the specified tags. Cannot be used with <see cref="WithRules(string[])"/>
+        /// </summary>
+        /// <param name="tags">tags to be used for scanning</param>
+        public AxeBuilder WithTags(params string[] tags)
+        {
+            ValidateParameters(tags, nameof(tags));
+
+            Options = null;
+            runOptions.RunOnly = new RunOnlyOptions
+            {
+                Type = "tag",
+                Values = tags.ToList()
+            };
+            return this;
+        }
+
+        /// <summary>
+        /// Limit analysis to only the specified rules. Cannot be used with <see cref="WithTags(string[])"/>
```

Comment:
The target audience probably won't know offhand which rules are valid. Maybe we should note how to find the IDs in the docs?

```
See https://dequeuniversity.com/rules/axe/ for a complete listing of available rules IDs.
```
```
Commonly used tags include "wcag2a", "wcag2aa", "wcag21aa", "best-practice", and "experimental".

See https://www.deque.com/axe/axe-for-web/documentation/api-documentation/#api-name-axegetrules for complete descriptions of all available tags.
```